Migration to csproj. Added sample app to test the code

cmatskas opened the pull request:
As per our discussion on issue #29, I've implemented the necessary changes to migrate to the new csproj format to support .NETStandard and the latest project format. Some of the changes in the project:
- csproj migration
- added sample project to test the code
- up'ed the version of AppInsights SDK to 2.2

Thanks @jbattermann  for the support

cmatskas commented:
@nblumhardt @jbattermann  any chance you could kick off another AppVeyor build and see if we can get this working? Did I miss something on my end? Thanks

jbattermann commented:
@cmatskas Sorry I was on a business trip the past week and didn't get a chance to look closer _but_ according to the [build log at AppVeyour](https://ci.appveyor.com/project/serilog/serilog-sinks-applicationinsights/build/74) in line [42](https://ci.appveyor.com/project/serilog/serilog-sinks-applicationinsights/build/74#L42), it seems like there's a discrepancy / problem with the MS AI NuGet package referenced not being .Net Standard 1.3 compatible:

```
  Errors in C:\projects\serilog-sinks-applicationinsights\src\Serilog.Sinks.ApplicationInsights\Serilog.Sinks.ApplicationInsights.csproj
      Package Microsoft.ApplicationInsights 2.2.0 is not compatible with netstandard1.3 (.NETStandard,Version=v1.3). Package Microsoft.ApplicationInsights 2.2.0 supports:
        - net40 (.NETFramework,Version=v4.0)
        - net45 (.NETFramework,Version=v4.5)
        - net46 (.NETFramework,Version=v4.6)
        - netstandard1.5 (.NETStandard,Version=v1.5)
        - portable-win81+wpa81 (.NETPortable,Version=v0.0,Profile=Profile32)
        - uap10.0 (UAP,Version=v10.0)
        - wp8 (WindowsPhone,Version=v8.0)
      One or more packages are incompatible with .NETStandard,Version=v1.3.
 
```

Not sure if the MS NuGet has a 1.3 compatible (pre-rel?) one or otherwise I guess the sink needs to follow suit in regards to min .Net Standard supported..

jbattermann commented on serilog-sinks-applicationinsights.sln:
```diff
@@ -14,7 +14,11 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "assets", "assets", "{E9D1B5
-Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Serilog.Sinks.ApplicationInsights", "src\Serilog.Sinks.ApplicationInsights\Serilog.Sinks.ApplicationInsights.xproj", "{B572E129-3568-435F-A8E9-366AD7014D8E}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Serilog.Sinks.ApplicationInsights", "src\Serilog.Sinks.ApplicationInsights\Serilog.Sinks.ApplicationInsights.csproj", "{B572E129-3568-435F-A8E9-366AD7014D8E}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Sample", "Sample", "{79D27B67-A40C-4078-B4C9-A981BADB544B}"
```
I'd keep adding a new (sample) project (the AppInsightsDemo) out of this PR... while it is right that it should / could contain a few examples regarding usage of the sink, I'd track it as a separate issue/PR if still wanted/need.

cmatskas replied:
Ok, I will remove the sample and raise a separate issue to track this. Thanks for he feedback

jbattermann commented:
Otherwise the PR looks 'almost' good.. if it's ok with you I'd leave the sample project out of this PR though.. 👍

cmatskas commented:
Sample project will be separated and I'll recommit ;) thanks all for your help

cmatskas commented:
@jbattermann I've removed the sample project and updated the AppInsights version with the latest push. I hope that this is OK now and we can get this merged :) All checks have passed too!

jbattermann commented:
This took way too long - sorry @cmatskas but thanks so much for the work!
